lib/AST/Decl.cpp

```diff
@@ -2450,6 +2450,10 @@ DestructorDecl *ClassDecl::getDestructor() {
 }
 
 bool ClassDecl::inheritsSuperclassInitializers(LazyResolver *resolver) {
+  // Get a resolver from the ASTContext if we don't have one already.
+  if (resolver == nullptr)
+    resolver = getASTContext().getLazyResolver();
```

Comment:
it is not really clear that this is the intent and a lot of code just reaches for the resolver from the context. But if you insist, I can change it so that the caller of this method passes in the context's resolver instead of null. That's just NFC though. I'd rather rip out the threading around of the resolver altogether since it's essentially a dynamically scoped variable.